source/NVDAObjects/UIA/__init__.py

```diff
@@ -789,7 +789,15 @@ def findOverlayClasses(self,clsList):
 		elif UIAControlType==UIAHandler.UIA_ListItemControlTypeId:
 			clsList.append(ListItem)
 		# #5942: In Windows 10 build 14332 and later, Microsoft rewrote various dialog code including that of User Account Control.
-		if self.UIAIsWindowElement and UIAClassName in ("#32770","NUIDialog", "Credential Dialog Xaml Host"):
+		# #8405: there are more dialogs scattered throughout Windows 10 and various apps.
+		# Apart from "popup" dialog seen when uninstaling apps and such, they are window elements (popup dialog isn't).
+		# Dialog detection is a bit easier on build 17682 and later thanks to IsDialog property.
+		try:
+			isDialog = self._getUIACacheablePropertyValue(UIAHandler.UIA_IsDialogPropertyId)
+		except COMError:
+			isDialog = False

```

Comment:
Instead of setting this to False, you could consider setting isDialog to the expression in the if clause. like: 

`isDialog = (self.UIAIsWindowElement and UIAClassName in UIAHandler.UIADialogClassNames) or (not self.UIAIsWindowElement and UIAClassName == "Popup"):`

To me, this is a bit more readable. Or are there cases where isDialog returns False, and yet we should treat something as a dialog? In that case, the "not isdialog" check still seems redundant. I think we should always respect isDialog if possible.